src/service_aggregator.py

```diff
@@ -655,19 +655,29 @@ def merge_results_by_node(result_message, merge_qnode):
 
 async def robokop_infer(input_message, guid, question_qnode, answer_qnode):
     automat_url = os.environ.get("ROBOKOPKG_URL", "https://automat.transltr.io/robokopkg/1.3/")
+    max_conns = os.environ.get("MAX_CONNECTIONS", 30)
+    nrules = int(os.environ.get("MAXIMUM_ROBOKOPKG_RULES", 75))
     messages = expand_query(input_message, {}, guid)
+    logger.debug(f"{guid}: {len(messages)} to send to {automat_url}")
     result_messages = []
-    for message in messages:
+    limits = httpx.Limits(max_keepalive_connections=None, max_connections=max_conns)
```

Comment:
I believe you need to use this `limits` variable in the httpx AsyncClient. And it might not be as simple as adding it on line 667. We might need to create a `client` with the limit outside of the loop and then close it ourselves afterwards, but I'm not entirely sure.

Reply:
Ah yes, I found this difficult to test

Reply:
With some improved logging in plater, it's a bit easier to see what's going on.  I was never able to get the limits approach to really work, but this version based on a semaphore appears to do the right thing.